### PR TITLE
Fix mqtt connection

### PR DESCRIPTION
**Pull request by Kraemii**
**What**:
Don't exit with exception in case the MQTT connection is lost
Automatically subscribes again to all topics after a reconnect of MQTT
SC-649
GOS-1037

**Why**:
In case the ospd-openvas MQTT client disconnect from its broker, it exited with an exception due to a bug within the currently used paho-mqtt version. This issue is fixed with this PR

**How**:
Run ospd-openvas and notus-scanner. They should connect to a MQTT broker. After ospd-openvas is initialized stop the broker and restart it after a few seconds. Start a scan and watch for notus results. In the old version ospd-openvas should not be able to receive any. This PR fixes this issue. Note that ospd-openvas had a similar problem, which will get fixed with https://github.com/greenbone/ospd-openvas/pull/713
Start scan via gvm-cli:
```xml
<start_scan scan_id='97079ee9-8917-49da-aa4f-4ef95f757ac1'>
    <vt_selection>
        <vt_single id='1.3.6.1.4.1.25623.1.0.50282'>
        </vt_single>
    </vt_selection>
    <targets>
        <target>
            <hosts>localhost</hosts>
            <ports>80,8080,443,22</ports>
            <credentials>
                <credential service="ssh" type="up" port="22">
                    <username>ssh_user</username>
                    <password>ssh_pw</password>
                </credential>
            </credentials>
        </target>
    </targets>
    <scanner_params>
    </scanner_params>
</start_scan>
```

**Checklist**:

- [ ] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/notus-scanner/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -40,7 +40,7 @@ packages = [
 
 [tool.poetry.dependencies]
 python = "^3.7"
-paho-mqtt = "^1.6"
+paho-mqtt = "1.5.1"
```
Comment (bjoernricks) [inline pyproject.toml:43]: Please never stick with a specific version. This results in all kind of issues for the community.
```suggestion
paho-mqtt = ">=1.5.1"
```

**Review by bjoernricks — DISMISSED** [review]

File: pyproject.toml (hunk 1 above)
Comment (ArnoStiefvater) [reply pyproject.toml:43]: (same comment as in the ospd-openvas repo):

> Please support newer versions of paho-mqtt too just use >=1.5.1

Currently the version 1.5.1 is in debian bullseye which has some serious bugs we have to work around. The on_disconnect function is not working in 1.5.1 resulting in a crash when the broker disconnects. We want to be aware of this while developing so we do not run into such problems again.

**Review by ArnoStiefvater — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (ArnoStiefvater) [reply pyproject.toml]: We chose to do your suggestion of using ">=1.5.1" for now until a better solution is found.

Comment (bjoernricks) [reply pyproject.toml]: Thanks a lot!

**Pull request merged**